src/Microsoft.CodeAnalysis.Analyzers/VisualBasic/VisualBasicUpgradeMSBuildWorkspaceAnalyzer.vb

```diff
@@ -0,0 +1,49 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.Analyzers
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Analyzers
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Public Class VisualBasicUpgradeMSBuildWorkspaceAnalyzer
+        Inherits UpgradeMSBuildWorkspaceAnalyzer
+
+        Private Sub New(performAssemblyChecks As Boolean)
+            MyBase.New(performAssemblyChecks)
+        End Sub
+
+        Public Sub New()
+            Me.New(performAssemblyChecks:=True)
+        End Sub
+
+        Friend Shared Function CreateForTests() As VisualBasicUpgradeMSBuildWorkspaceAnalyzer
+            Return New VisualBasicUpgradeMSBuildWorkspaceAnalyzer(performAssemblyChecks:=False)
+        End Function
+
+        Protected Overrides Sub RegisterIdentifierAnalysis(context As CompilationStartAnalysisContext)
+            context.RegisterSyntaxNodeAction(AddressOf AnalyzerIdentifier, SyntaxKind.IdentifierName)
+        End Sub
+
+        Protected Overrides Sub RegisterIdentifierAnalysis(context As AnalysisContext)
+            context.RegisterSyntaxNodeAction(AddressOf AnalyzerIdentifier, SyntaxKind.IdentifierName)
+        End Sub
+
+        Private Sub AnalyzerIdentifier(context As SyntaxNodeAnalysisContext)
+            If TypeOf context.Node IsNot IdentifierNameSyntax Then
+                Return
+            End If
+
+            Dim identifierName = DirectCast(context.Node, IdentifierNameSyntax)
+
+            If identifierName.Identifier.ToString() <> MSBuildWorkspace Then
```

Comment:
I'm torn between pointing out this should be VB case insensitive, and just saying that those who do that deserve their own fate.

Reply:
yeah, of course it should be. Will fix.